src/Core/Renderer.php

```diff
@@ -66,28 +66,27 @@ class Renderer
 	];
 
 	/**
-	 * This is our template processor
+	 * Returns the rendered template output from the template string and variables
 	 *
-	 * @param string|FriendicaSmarty $s    The string requiring macro substitution or an instance of FriendicaSmarty
-	 * @param array                  $vars Key value pairs (search => replace)
-	 *
-	 * @return string substituted string
-	 * @throws Exception
+	 * @param string $template
+	 * @param array  $vars
+	 * @return string
+	 * @throws InternalServerErrorException
 	 */
-	public static function replaceMacros($s, array $vars = [])
+	public static function replaceMacros(string $template, array $vars)
```

Comment:
Are you sure that the function is always called with a second parameter?

Reply:
Nope, nice catch!